codes/quantum/qubits/subsystem/subsystem-gen-shor.yml

```diff
@@ -0,0 +1,46 @@
+
+protection: 'Detecting single qubit errors of weight \(min(d_1,d_2)\), this code corrects bit flip and phase flip errors modulo the subsystem structure.'
+
+features:
+
+  encoders:
+    - 'Let \(G_1\) and \(G_2\) be the generator matrices for \(\mathcal{C}_1\) and \(\mathcal{C}_2\) respectively, \(P_1\) and \(P_2\) be the parity check matrices for \(\mathcal{C}_1\) and \(\mathcal{C}_2\) respectively.'
+    - 'Construct stabilizer codes on \(n_1\) and \(n_2\) qubits respectively, using the rows of \(P_1\) and \(P_2\): \(\mathcal{S}_1\) is the stabilizer generated by \(\{S_i: S_i = \otimes_{j=1}^{n_1}Z^{(P_1)_{ij}}, 1 \leq i \leq n_1 - k_1\}\), and \(\mathcal{S}_2\) is the stabilizer generated by \(\{T_i: T_i = \otimes_{j=1}^{n_2}X^{(P_2)_{ij}}, 1 \leq i \leq n_2 - k_2\}\). Correspondingly, the first code corrects $\lfloor\frac{d_1-1}{2}\rfloor$ bit flip errors and the second code corrects \(\lfloor\frac{d_2-1}{2}\rfloor\) phase flip errors.'
+    - 'Let \(n = n_1 + n_2\). Consider a square lattice of size \(n \times n\) with qubits located at the vertices of this lattice. Let \(\mathcal{O}_{ij}\) denote the operator \(\mathcal{O}\) acting on the qubit located on the \(i\)th row and \(j\)th column of this lattice tensored with identity on all other qubits. Let \(\mathcal{T}_1\) be the stabilizer group made up of letting \(\mathcal{S}_1\) operators acting on all of the columns of the lattice. Let \(\mathcal{T}_2\) be the stabilizer group made up of letting \(\mathcal{S}_2$\) operators acting on all of the rows of the lattice.'
+    - '\(\mathcal{T} = \langle \mathcal{T}_1, \mathcal{T}_2\rangle\) consists of Pauli operators which have an even number of \(X_{i,j}\) operators in each column, and an even number of \(Z_{i,j}\) operators in each row. \(\mathcal{T}\) is nonabelian. Let \(\mathcal{S}\) be an abelian subgroup of \(\mathcal{T}\). It consists of Pauli operators which are made up of an even number of rows consisting entirely of \(X\) operators and an even number of columns consisting entirely of \(Z\) operators. Let \(\mathcal{L}\) be the set of logical operators who commute with elements of \(\mathcal{T}\) and \(\mathcal{S}\).'
+    - 'To form a subsystem code using \(\mathcal{T}\), \(\mathcal{S}\), and \(\mathcal{L}\), note that there are $(n_1-k_1)k_2$ stabilizer generators made up of tensor product of $I$ and $Z$ operators and $(n_2-k_2)k_1$ stabilizer generators made up of tensor product of $I$ and $X$ operators. In total, there are $(n_1-k_1)k_2+(n_2-k_2)k_1$ stabilizer generators. Then proceed following \cite{arXiv:quant-ph/0506023v4}.'
+
+  decoders:
+    - 'By measuring the generators of \(\mathcal{S}\), we can correct errors related to the classical linear codes \(\mathcal{C}_1\) and \(\mathcal{C}_2\).'
+
+notes:
```

Comment:
realizations:
- 'Realized in all-photonic quantum repeater architecture \cite{arxiv:2203.07979}.'